src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+            sourceRepo: isForwardFlow ? productRepoForkUri : VmrForkUri,
+            targetRepo: isForwardFlow ? VmrForkUri : productRepoForkUri,
+            targetBranch: isForwardFlow ? vmrTmpBranch.Value : subscription.TargetBranch,
+            sourceDirectory: subscription.SourceDirectory,
+            targetDirectory: subscription.TargetDirectory);
+
+        // Add it to the channel
+        await darcProcessManager.AddBuildToChannelAsync(build.Id, channelName);
+
+        // Trigger the subscription
+        await TriggerSubscriptionAsync(testSubscription.Value);
+
+        logger.LogInformation("Code flow successfully recreated. Press enter to finish and cleanup");
+        Console.ReadLine();
+
+        // Cleanup
+        await DeleteAllDarcPRBranchesAsync(isForwardFlow ? VmrForkRepoName : productRepoUri.Split('/').Last());
+    }
+
+    private async Task DeleteAllDarcPRBranchesAsync(string repo)
+    {
+        var branches = await ghClient.Repository.Branch.GetAll(MaestroAuthTestOrgName, repo);
+        foreach (var branch in branches.Where(b => b.Name.StartsWith(DarcPRBranchPrefix)))
+        {
+            await DeleteGitHubBranchAsync(repo, branch.Name);
+        }
+    }
+
+    private async Task AddRepositoryToBarIfMissingAsync(string repositoryName)
+    {
+        if ((await context.Repositories.FirstOrDefaultAsync(repo => repo.RepositoryName == repositoryName)) == null)
+        {
+            context.Repositories.Add(new Maestro.Data.Models.Repository
+            {
+                RepositoryName = repositoryName,
+                InstallationId = InstallationId
+            });
+            await context.SaveChangesAsync();
+        }
+    }
+
+    private async Task<Build> CreateBuildAsync(string repositoryUrl, string branch, string commit)
+    {
+        Build build = await pcsApi.Builds.CreateAsync(new BuildData(
+            commit: commit,
+            azureDevOpsAccount: "test",
+            azureDevOpsProject: "test",
+            azureDevOpsBuildNumber: "1",
+            azureDevOpsRepository: repositoryUrl,
+            azureDevOpsBranch: branch,
+            released: false,
+            stable: false)
+        {
+            GitHubRepository = repositoryUrl,
+            GitHubBranch = branch
+        });
+
+        return build;
+    }
+
+    private async Task TriggerSubscriptionAsync(string subscriptionId)
+    {
+        await pcsApi.Subscriptions.TriggerSubscriptionAsync(default, Guid.Parse(subscriptionId));
+    }
+
+    private async Task<AsyncDisposableValue<string>> PrepareVmrForkAsync(
+        string branch,
+        string productRepoUri,
+        string productRepoForkUri)
+    {
+        // Sync the VMR fork branch
+        await SyncForkAsync("dotnet", "dotnet", branch);
+        // Check if the user has the forked VMR in local DB
+        await AddRepositoryToBarIfMissingAsync(VmrForkUri);
+
+        // Create a temporary branch
+        var newBranchName = Guid.NewGuid().ToString();
```

Comment:
Please put some prefix like `repro/` or something that will allow us to identify/manage all of the branches easily.
Many git viewers clump branches together by common prefixes so it's quite useful.

Reply:
added